src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,197 @@
+import {Trans} from '@lingui/react'
+import Color from 'color'
+import {DataLink} from 'components/ui/DbLink'
+import {ActionKey} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const SOUL_MAX_VALUE = 100
+const SHROUD_MAX_VALUE = 100
+
+const SOUL_GAUGE_COLOR = Color('#DC0707FF')
+const SHROUD_GAUGE_COLOR = Color('#2C9FCB')
+
+const SOUL_GENERATORS: ActionKey[] = [
+	'SLICE',
+	'WAXING_SLICE',
+	'INFERNAL_SLICE',
+	'SPINNING_SCYTHE',
+	'NIGHTMARE_SCYTHE',
+	'SOUL_SLICE',
+	'SOUL_SCYTHE',
+]
+
+const SOUL_CONSUMERS: ActionKey[] = [
+	'BLOOD_STALK',
+	'GRIM_SWATHE',
+	'GLUTTONY',
+	'UNVEILED_GIBBET',
+	'UNVEILED_GALLOWS',
+]
+
+const SHROUD_GENERATORS: ActionKey[] = [
+	'GIBBET',
+	'GALLOWS',
+	'GUILLOTINE',
+	'PLENTIFUL_HARVEST',
+]
```

Comment:
same with this and Plentiful Harvest, it could even just be on its own in a hook, since it's just the one action.